aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -184,6 +184,15 @@ enum UpdateResult {
 }
 
 impl<T: Transaction> CapturedReads<T> {
+    // Return an iterator over the captured reads.
+    pub(crate) fn get_read_values_with_delayed_fields(
+        &self,
+    ) -> impl Iterator<Item = (&T::Key, &DataRead<T::Value>)> {
```

Comment:
can we map to pair of (&V, &MoveTypeLayout), instead of leaking DataRead?

that way we can have the sequential code provide the same iterator potentially?